SingularityExecutor/src/main/java/com/hubspot/singularity/executor/config/SingularityExecutorLogrotateAdditionalFile.java

```diff
@@ -3,35 +3,43 @@
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.google.common.base.Optional;
+import com.hubspot.singularity.executor.SingularityExecutorLogrotateFrequency;
 
 public class SingularityExecutorLogrotateAdditionalFile {
     private final String filename;
-    private final Optional<String> extension;
-    private final Optional<String> dateformat;
+    private final String extension;
+    private final String dateformat;
+    private final SingularityExecutorLogrotateFrequency frequencyOverride;
 
     @JsonCreator
     public static SingularityExecutorLogrotateAdditionalFile fromString(String value) {
-        return new SingularityExecutorLogrotateAdditionalFile(value, Optional.<String>absent(), Optional.<String>absent());
+        return new SingularityExecutorLogrotateAdditionalFile(value, null, null, null);
     }
 
     @JsonCreator
     public SingularityExecutorLogrotateAdditionalFile(@JsonProperty("filename") String filename,
-        @JsonProperty("extension") Optional<String> extension,
-        @JsonProperty("dateformat") Optional<String> dateformat) {
+        @JsonProperty("extension") String extension,
+        @JsonProperty("dateformat") String dateformat,
+        @JsonProperty("logrotateFrequencyOverride") SingularityExecutorLogrotateFrequency frequencyOverride) {
         this.filename = filename;
         this.extension = extension;
         this.dateformat = dateformat;
+        this.frequencyOverride = frequencyOverride;
     }
 
     public String getFilename() {
         return filename;
     }
 
     public Optional<String> getExtension() {
```

Comment:
Each of the getters in a POJO like this should have the same return type as the private member variables they expose. In this case, I'd recommend having `extension`, `dateformat`, and `frequencyOverride` be `Optional<>`s (in both their member variable types and their getter types).

Our Jackson YAML deserialization should automatically take care of setting a missing field to an empty `Optional<>` in the resulting Java object.